### PR TITLE
[DEV-2722] pull column descriptions from DWH

### PR DESCRIPTION
**Pull request by vikua**
## Description

Pull column descriptions from DFH comment fields. 

## Related Issue

[DEV-2722](https://featurebyte.atlassian.net/browse/DEV-2722)

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

[DEV-2722]: https://featurebyte.atlassian.net/browse/DEV-2722?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by vikua** [comment]
@gee-senbong can you please review? 
There are some weird changes in tests which I suppose come from hashing of ColumnSpec with description now.

File: featurebyte/query_graph/node/schema.py [hunk 1]
```diff
@@ -250,3 +250,4 @@ class ColumnSpec(FeatureByteBaseModel):
 
     name: StrictStr
     dtype: DBVarType
+    description: Optional[str]
```
Comment (gee-senbong) [inline featurebyte/query_graph/node/schema.py:253]: This schema is used in query graph (https://github.com/featurebyte/featurebyte/blob/d21ce4a278e0f9607b5a7e6e1d343ce24db33682/featurebyte/query_graph/node/input.py#L41).

If we add the parameter here, it means changing of the column description will cause change in query graph. `ColumnInfo` already contains `description` attribute, curious why introduces the parameter here (https://github.com/featurebyte/featurebyte/blob/0660cbdc426a811681753fb11ab8f88d9e95e968/featurebyte/query_graph/model/column_info.py#L51).

File: featurebyte/service/feature_store_warehouse.py [hunk 2]
```diff
@@ -202,4 +202,7 @@ async def list_columns(
         except db_session.no_schema_error as exc:
             raise TableNotFoundError(f"Table {table_name} not found.") from exc
 
-        return [ColumnSpec(name=name, dtype=dtype) for name, dtype in table_schema.items()]
+        return [
+            ColumnSpec(name=name, dtype=schema.dtype, description=schema.description)
+            for name, schema in table_schema.items()
+        ]
```
Comment (gee-senbong) [inline featurebyte/service/feature_store_warehouse.py:208]: We could add a new schema like `ColumnSpecWithDescription` at `featurebyte/query_graph/model/column_info.py`, and then use the schema here without modifying existing `ColumnSpec` schema. By doing so, we could avoid modifying underlying query graph (should not see lots of changes due to changes in hash values).

File: featurebyte/session/base_spark.py [hunk 3]
```diff
@@ -239,20 +244,27 @@ async def list_table_schema(
...
-                    var_info.upper()
+
+                dtype = self._convert_to_internal_variable_type(var_info.upper())
+                column_name_type_map[column_name] = ColumnSchema(
+                    name=column_name,
+                    dtype=dtype,
+                    description=comment,
                 )
+
         return column_name_type_map
```
Comment (gee-senbong) [inline featurebyte/session/base_spark.py:268]: This implementation makes sense to me. 👍

File: featurebyte/service/feature_store_warehouse.py [hunk 4]
```diff
@@ -159,7 +159,7 @@ async def list_columns(
         database_name: str,
         schema_name: str,
         table_name: str,
-    ) -> List[ColumnSpec]:
+    ) -> List[ColumnInfo]:
```
Comment (gee-senbong) [inline featurebyte/service/feature_store_warehouse.py:162]: Should we use `ColumnSchema` in `featurebyte/session/base.py` as `entity_id`, `critical_data_info` and `semantic_id` will be always empty?

**Review by gee-senbong — APPROVED** [review]
LGTM with a minor comment. Would be good to fix the schema issue in `featurebyte/service/feature_store_warehouse.py` before merging the PR.

File: featurebyte/service/feature_store_warehouse.py (hunk 4 above)
Comment (vikua) [reply featurebyte/service/feature_store_warehouse.py:162]: We do use ColumnSchema in session/base.py
Do you mean return in here as well?

Comment (gee-senbong) [reply featurebyte/service/feature_store_warehouse.py:162]: Yes. Sorry for the confusing comment, edited the comment to make it clearer.

**Pull request merged**